### PR TITLE
Integration with Syslog

### PR DESCRIPTION
**Pull request by dughetti**
@maxicecilia @lsvx I added a new service `Logger`. Now we can log directly to syslog by doing:  
```
Logger.log("Test");
Logger.info("Info");
Logger.error("Error");
```
We are using the npm module `ain2`. You can read the documentation on https://www.npmjs.com/package/ain2. I have also changed the way we response from the controllers. Instead of doing, for example, `res.json(500, { error: 'INVALID REQUEST' });`, now we are using sails responses methods, `return res.serverError({ error: 'INVALID REQUEST' }, err);`

### CODE REVIEW

File: api/controllers/ObjectController.js [hunk 1]
```diff
@@ -33,13 +33,16 @@ module.exports = {
             'RETURN object'
         ];
 
+        Logger.info('ObjectController: GET All objects of type: ', req.url, "Query: ", q.join('\n'));
+
         Activity.query(q, {}, function(err, results) {
             if (err) {
-                res.json(500, { error: 'INVALID REQUEST' });
+                return res.serverError({ error: 'INVALID REQUEST' }, err);
```
Comment (lsvx) [inline api/controllers/ObjectController.js]: Nice. These really messed us up last time.

File: api/responses/badRequest.js [hunk 2]
```diff
@@ -26,39 +26,11 @@ module.exports = function badRequest(data, options) {
     res.status(400);
 
     // Log error to console
-    if (data !== undefined) {
-      sails.log.verbose('Sending 400 ("Bad Request") response: \n',data);
+    if (err !== undefined) {
+      Logger.error('Sending 400 ("Bad Request") response: \n',JSON.stringify(err));
```
Comment (lsvx) [inline api/responses/badRequest.js]: good!

File: api/responses/badRequest.js [hunk 3]
```diff
@@ -3,19 +3,19 @@
...
- *   'Please choose a valid `password` (6-12 characters)',
- *   'trial/signup'
+ *   'Bad Request',
+ *   'Authentication policy: userId is null'
  * );
  * ```
  */
 
-module.exports = function badRequest(data, options) {
+module.exports = function badRequest(data, err) {
```
Comment (lsvx) [inline api/responses/badRequest.js]: Do you think we really want to ignore `data` entirely?

Comment (lsvx) [inline api/responses/badRequest.js]: nvm, you still do `res.send(data)`.

**Comment by lsvx** [comment]
This looks great, @dughetti!

**Comment by dughetti** [comment]
@lsvx @maxicecilia I removed unnecessary info logs. It will write to Syslog only errors. (cache error, bad responses, failed policy, etc.) If we are in production, It will show only error or warning logs in console.

**Comment by maxicecilia** [comment]
+1

**Pull request merged**